### DISCUSSION TITLE
LOGIN_TIMEOUT logs out the user from login instead of inactivity.

### DISCUSSION THREAD

**Discussion by abhishektalwar** [netbox-community/netbox#6868] [Q&A] [answered]
Hi Everyone,

I recently changed the LOGIN_TIMEOUT to 15 minutes (900 seconds) for my netbox setup. However, it does seem the logout is 15 minutes from login instead of 15 minutes of inactivity.

Is there a way/mechanism to set this so that it is 15 minutes of inactivity?

Looking forward to hearing from you.

**Comment by abhishektalwar** [comment]
@jeremystretch

**Reply by jeremystretch** [reply]
Please don't ping me for no reason.

**Reply by abhishektalwar** [reply]
Hi Jeremy, My apologies for pinging you like that. However, I am trying to find an answer for this issue and thought you will be right person to reach out. 
I am looking for a way to logout in case of inactivity.

**Reply by abhishektalwar** [reply]
The LOGIN_TIMEOUT in configuration.py logout the user even if he is active. However, I am trying to check if we can do something to logout a user only if he is inactive

**Reply by candlerb** [reply]
Since Netbox is written in Django, your google search would be something like "django timeout inactive login"

First hit I get: https://stackoverflow.com/questions/3024153/how-to-expire-session-due-to-inactivity-in-django

**Reply by abhishektalwar** [reply]
@candlerb 
Tried the following, however it didn't work:-

**SESSION_EXPIRE_AT_BROWSER_CLOSE = True
SESSION_COOKIE_AGE = 900
SESSION_SAVE_EVERY_REQUEST = True**

**Reply by candlerb** [reply]
Which file did you put them in?

I believe it would have to be in settings.py, since there's no corresponding value in configuration.py - except for SESSION_COOKIE_AGE which is set from Netbox's configuration LOGIN_TIMEOUT [here](https://github.com/netbox-community/netbox/blob/v2.11.10/netbox/netbox/settings.py#L253) - previously picked up [here](https://github.com/netbox-community/netbox/blob/v2.11.10/netbox/netbox/settings.py#L95).

You shouldn't normally modify files which belong to Netbox.  However if you find this does what you need, you can raise a feature request for a new variable in configuration.py

**Comment by koratfood** [comment, marked as answer]
Ref. #6929, the recently-released v2.11.12 introduces an optional configuration parameter called `LOGIN_PERSISTENCE`. Setting it to True will cause Netbox to behave the way you want. Make sure you read the caveats in the example config first, though.
